plugins/woocommerce/includes/wc-formatting-functions.php

```diff
@@ -1225,11 +1225,21 @@ function wc_format_option_price_num_decimals( $value, $option, $raw_value ) {
 function wc_format_option_hold_stock_minutes( $value, $option, $raw_value ) {
 	$value = ! empty( $raw_value ) ? absint( $raw_value ) : ''; // Allow > 0 or set to ''.
 
-	wp_clear_scheduled_hook( 'woocommerce_cancel_unpaid_orders' );
+	// Clear existing scheduled events.
+	as_unschedule_all_actions( 'woocommerce_cancel_unpaid_orders' );
 
 	if ( '' !== $value ) {
+		/**
+		 * Filters the interval at which to cancel unpaid orders in minutes.
+		 *
+		 * @since 5.1.0
+		 *
+		 * @param int $cancel_unpaid_interval The interval at which to cancel unpaid orders in minutes.
+		 */
 		$cancel_unpaid_interval = apply_filters( 'woocommerce_cancel_unpaid_orders_interval_minutes', absint( $value ) );
-		wp_schedule_single_event( time() + ( absint( $cancel_unpaid_interval ) * 60 ), 'woocommerce_cancel_unpaid_orders' );
+
+		// Schedule the next event using Action Scheduler if available, otherwise fall back to WordPress cron.
+		as_schedule_single_action( time() + ( absint( $cancel_unpaid_interval ) * 60 ), 'woocommerce_cancel_unpaid_orders' );
```

Comment:
This action does not have a group so results in duplicate entries in the table, is this intentional?

<img width="1510" alt="image" src="https://github.com/user-attachments/assets/448aa740-9130-42fc-a433-33ea4ec03c85" />

Looks like it will keep re-adding itself too in the `woocommerce_cancel_unpaid_orders` function, so we'll have one scheduled in the `woocommerce` group from `register_recurring_actions` and then another one perpetually added with no group from  `woocommerce_cancel_unpaid_orders`

I know this was the case before, but do we want to look at it?